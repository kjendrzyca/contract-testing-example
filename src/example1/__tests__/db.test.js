// Glossary:
// ğŸ’¡ - a note
// ğŸ¦– - an action
// ---

// ğŸ’¡ for simulating database changes over time
const db1 = require('../db1')
const db2 = require('../db2')
const db3 = require('../db3')
// ---

const createApi = require('../api')

describe.skip('db test', () => {
  it('fetches all movies', async () => {
    // given
    let api

    {
      // ğŸ¦– 1) run test once and comment out:
      // ğŸ‘‰ check the snapshot
      api = createApi(db1)
    }

    {
      // ğŸ¦– 2) uncomment after the first run and run again
      // ğŸ‘‰ the test should turn red
      // ğŸ‘‰ fix it
      // ğŸ‘‰ check the snapshot
      // api = createApi(db2)
    }

    {
      // ğŸ¦– 3) uncomment after the second run and run again
      // ğŸ‘‰ the test should turn red
      // ğŸ‘‰ fix it
      // api = createApi(db3)
    }

    // when
    const movies = await api.getAll()

    // then
    expect(movies).toMatchSnapshot();

    {
      // ğŸ¦– 4) update the db and try running again
      // ğŸ‘‰ it won't work
      // ğŸ‘‰ remove the assertion above
      // ğŸ‘‰ add custom matcher to make it work again
    }
  })
})
