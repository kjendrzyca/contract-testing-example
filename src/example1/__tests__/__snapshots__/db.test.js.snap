// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db test fetches all movies - db1 1`] = `
Object {
  "director": "Christopher Nolan",
  "id": 1,
  "lastFetchDate": Any<Date>,
  "name": "Interstellar",
  "runtime": 169,
  "year": "2014",
}
`;
